apps/backend/src/users/users.service.ts

```diff
@@ -1,17 +1,55 @@
-import { Injectable } from '@nestjs/common';
+import {
+  BadRequestException,
+  UnauthorizedException,
+  Injectable,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { Repository } from 'typeorm';
+import { MongoRepository } from 'typeorm';
 
 import { User } from './user.entity';
+import { Status } from './types';
+import { ObjectId } from 'mongodb';
 
 @Injectable()
 export class UsersService {
   constructor(
     @InjectRepository(User)
-    private usersRepository: Repository<User>,
+    private usersRepository: MongoRepository<User>,
   ) {}
 
-  findAll(): Promise<User[]> {
-    return this.usersRepository.find();
+  async findAll(getAllMembers: boolean): Promise<User[]> {
+    if (!getAllMembers) return [];
+
+    const exampleUser: User = {
+      userId: new ObjectId('a0f3efa0f3efa0f3efa0f3ef'),
+      status: Status.ADMIN,
+      firstName: 'jimmy',
+      lastName: 'jimmy2',
+      email: 'jimmy.jimmy2@mail.com',
+      profilePicture: null,
+      linkedin: null,
+      github: null,
+      team: null,
+      role: null,
+    };
+
+    if (exampleUser.status == Status.APPLICANT) {
+      throw new UnauthorizedException();
+    }
+
+    const users: User[] = await this.usersRepository.find({
+      where: {
+        status: { $not: { $eq: Status.APPLICANT } },
+      },
+    });
+
+    //TODO possibly use an interceptor instead here
+    users.forEach((user: User) => {
```

Comment:
:question: Is this check necessary since all of the users whose status is `Applicant` should've been filtered out by line 42? When can the condition on line 48 be true?

Reply:
Not really, we just included it because the ticket said to have it there. I guess the idea was that this check is just to guarantee that line 42 does what it's supposed to. 
From a more theoretical standpoint, I think it makes more sense to have this check defined outside this service and simply applied to the return value from an arbitrary endpoint where this check needs to be applied. This would act as a data validation check in case we change our code in the future (for example, messing with line 42 so that it accidently includes applicants).

Reply:
Gotcha, that makes sense! One quick thing I just wanted to mention is that I think it's important to push back on people's ideas if you think it's not the best. When working on a team, there's always going to be times when you don't agree with something someone suggests and I think it's good to get clarity on their thoughts when that happens (and when I say "push back on people's ideas", I don't mean waging some kind of war against them or something but just reaching out to them and asking why certain decisions were made haha). I've personally been asked questions like "What do you do when a decision is made that you don't agree with?" or even just straight up "Tell me about a time when you had to push back on other people's ideas. Whose idea was it and how did you do it?" in a few behavioral interviews before so even in the real world, people (or at least the _good_ kind of people) expect their coworkers to raise concerns about their ideas if they have any

As for having the applicant check outside of the service, I think that could definitely make sense if it were something that multiple routes could take advantage of. But since this check is fairly specific to this route (because in order for this check to be useful in other routes, the route needs to 1) return one or more users, and 2) must explicitly reject users of certain types from being returned), I think it's ok for this to not be abstracted out. It's really good that you were thinking about the code changing and possibly breaking in the future (because I've definitely done that before lol) but I'd say that should be handled by writing unit and/or integration tests because their purpose is to check for regression (i.e. code changes in the future and now something is behaving differently)